File: strings/middle.js


const middle = (s1, s2) => {
    let m = Math.floor(s1.length / 2);
    let [pre, post] = [s1.slice(0, m), s1.slice(m)];

    return pre + s2 + post;
}

let s1 = "Ault";
let s2 = "Kelly";

console.log(middle(s1, s2));

File: strings/middle.py
import math 

def middle(s1, s2):
    m = int(math.floor((len(s1) / 2))) 
    pre, post = s1[:m], s1[m:]

    return pre + s2 + post 

s1 = "Ault"
s2 = "Kelly"

print(middle(s1, s2))